cfgmgr/vlanmgr.cpp

```diff
@@ -669,6 +674,307 @@ void VlanMgr::doVlanMemberTask(Consumer &consumer)
+            }
+            SWSS_LOG_NOTICE("Add Vlan Member key: %s", kfvKey(t).c_str());
+            if (addHostVlanMember(vlan_id, port_alias, tagging_mode))
+            {
+                key = VLAN_PREFIX + to_string(vlan_id);
+                key += DEFAULT_KEY_SEPARATOR;
+                key += port_alias;
+                vector<FieldValueTuple> fvVector = kfvFieldsValues(t);
+                FieldValueTuple s("dynamic", "yes");
+                fvVector.push_back(s);
+                m_appVlanMemberTableProducer.set(key, fvVector);
+
+                vector<FieldValueTuple> fvVector1;
+                FieldValueTuple s1("state", "ok");
+                fvVector.push_back(s1);
+                m_stateVlanMemberTable.set(kfvKey(t), fvVector);
+            }
+            else
+            {
+                it++;
+                continue;
+            }
+        }
+        else if (op == DEL_COMMAND)
+        {
+            if (isVlanMemberStateOk(kfvKey(t)))
+            {
+                SWSS_LOG_NOTICE("Remove Vlan Member key: %s", kfvKey(t).c_str());
+                removeHostVlanMember(vlan_id, port_alias);
+                key = VLAN_PREFIX + to_string(vlan_id);
+                key += DEFAULT_KEY_SEPARATOR;
+                key += port_alias;
+                m_appVlanMemberTableProducer.del(key);
+                m_stateVlanMemberTable.del(kfvKey(t));
+            }
+            else
+            {
+                SWSS_LOG_DEBUG("%s doesn't exist", kfvKey(t).c_str());
+            }
+            auto vlans = m_PortVlanMember[port_alias];
+            for (const auto& vlan : vlans)
+            {
+                string vlan_alias = vlan.first;
+                string tagging_mode = vlan.second;
+                SWSS_LOG_NOTICE("Add Vlan Member vlan: %s port %s tagging_mode %s",
+                    vlan_alias.c_str(), port_alias.c_str(), tagging_mode.c_str());
+                addPortToVlan(port_alias, vlan_alias, tagging_mode);
+            }
+        }
+        else
+        {
+            SWSS_LOG_ERROR("Unknown operation type %s", op.c_str());
+        }
+        /* Other than the case of member port/lag is not ready, no retry will be performed */
+        it = consumer.m_toSync.erase(it);
+    }
+}
+
+void VlanMgr::addPortToVlan(const std::string& membername, const std::string& vlan_alias,
+        const std::string& tagging_mode)
+{
+    SWSS_LOG_NOTICE("member %s vlan %s tagging_mode %s",
+        membername.c_str(), vlan_alias.c_str(), tagging_mode.c_str());
+    int vlan_id = stoi(vlan_alias.substr(4));
+    if (addHostVlanMember(vlan_id, membername, tagging_mode))
+    {
+        std::string key = VLAN_PREFIX + to_string(vlan_id);
+        key += DEFAULT_KEY_SEPARATOR;
+        key += membername;
+        vector<FieldValueTuple> fvVector;
+        FieldValueTuple s("tagging_mode", tagging_mode);
+        fvVector.push_back(s);
+        FieldValueTuple s1("dynamic", "no");
```

Comment:
what does the "dynamic" field indicate here?

Reply:
"dynamic" means that the vlan member was not added by config, but was added by an application. This field can be used to indicate non-configured vlan from show commands.